src/Serialization/Rings.jl

```diff
@@ -226,19 +226,31 @@ end
 @registerSerializationType(nmod_mat)
 
 function save_internal(s::SerializerState, m::MatrixElem)
+    m_base_ring = save_type_dispatch(s, base_ring(parent(m)))
     return Dict(
         :matrix => save_type_dispatch(s, Array(m)),
+        :base_ring => m_base_ring
```

Comment:
the reason i run save_type_dispatch outside is so that it runs first, running it inside Dict does not guarantee the order, and it is important for backref purposes to have the base_ring outside

Reply:
You are not running it "in" the dict, but rather in the argument list of the function `Dict`. While it is true that the order of key-value pairs in a dictionary is arbitrary, the evaluation order of function argument is still fixed and fully deterministic.